core/src/main/java/com/linecorp/armeria/common/Cookie.java

```diff
@@ -99,6 +99,90 @@ static Cookies fromCookieHeader(boolean strict, String cookieHeader) {
+     *
+     * @param cookieHeaders the {@code "Cookie"} header values.
+     * @return the decoded {@link Cookie}s.
+     */
+    static Cookies fromCookieHeaders(String... cookieHeaders) {
+        return fromCookieHeaders(true, cookieHeaders);
+    }
+
+    /**
+     * Decodes the specified {@code "Cookie"} header values into a set of {@link Cookie}s.
+     *
+     * @param cookieHeaders the {@code "Cookie"} header values.
+     * @return the decoded {@link Cookie}s.
+     */
+    static Cookies fromCookieHeaders(Iterable<String> cookieHeaders) {
+        return fromCookieHeaders(true, cookieHeaders);
+    }
+
+    /**
+     * Decodes the specified {@code "Cookie"} header values into a set of {@link Cookie}s.
+     *
+     * @param strict whether to validate that the cookie names and values are in the valid scope
+     *               defined in RFC 6265.
+     * @param cookieHeaders the {@code "Cookie"} header values.
+     * @return the decoded {@link Cookie}s.
+     */
+    static Cookies fromCookieHeaders(boolean strict, String... cookieHeaders) {
+        requireNonNull(cookieHeaders, "cookieHeaders");
+        if (cookieHeaders.length == 0) {
+            return Cookies.of();
+        }
+
+        final ImmutableSet.Builder<Cookie> builder =
+                ImmutableSet.builderWithExpectedSize(cookieHeaders.length);
+        for (String v : cookieHeaders) {
+            requireNonNull(v, "cookieHeaders contains null.");
+            final Cookies cookies = fromCookieHeader(strict, v);
+            builder.addAll(cookies);
+        }
+
+        return Cookies.of(builder.build());
+    }
+
+    /**
+     * Decodes the specified {@code "Cookie"} header values into a set of {@link Cookie}s.
+     *
+     * @param strict whether to validate that the cookie names and values are in the valid scope
+     *               defined in RFC 6265.
+     * @param cookieHeaders the {@code "Cookie"} header values.
+     * @return the decoded {@link Cookie}s.
+     */
+    static Cookies fromCookieHeaders(boolean strict, Iterable<String> cookieHeaders) {
+        if (cookieHeaders instanceof Collection) {
+            return fromCookieHeaders(strict, (Collection<String>) cookieHeaders);
+        }
+
+        requireNonNull(cookieHeaders, "cookieHeaders");
+        final Iterator<String> it = cookieHeaders.iterator();
+        if (!it.hasNext()) {
+            return Cookies.of();
+        }
+
+        return CookieUtil.fromCookieHeaders(ImmutableSet.builder(), strict, it);
+    }
+
+    /**
+     * Decodes the specified {@code "Cookie"} header values into {@link Cookie}s.
+     *
+     * @param strict whether to validate the cookie names and values are in the valid scope defined in RFC 6265.
+     * @param cookieHeaders the {@code "Cookie"} header values.
+     * @return the decoded {@link Cookie}s.
+     */
+    static Cookies fromCookieHeaders(boolean strict, Collection<String> cookieHeaders) {
```

Comment:
We usually have two versions that are `String... cookieHeaders` and `Iterable<String> cookieHeaders` when the parameter type is a collection.
So could you remove this method, please?